master/buildbot/process/build.py

```diff
@@ -325,7 +325,10 @@ def startBuild(self, build_status, workerforbuilder):
         # If it returns failure then we don't start a new build.
         if ready_or_failure is not True:
             yield self.buildPreparationFailure(ready_or_failure, "worker_prepare")
-            self.buildFinished(["worker", "not", "available"], RETRY)
```

Comment:
well. I think the cancel should be instant, it does not have to wait until timeout, which can be quite long.

I think stopBuild() should force stop of instantiation somehow, probably by forcing the timer to timeout.